primitives/src/lib.rs

```diff
@@ -16,8 +19,8 @@ pub mod v0 {
 	mod error {
 		use super::*;
 
-		/// Reason why a Pop API call failed.
-		#[derive(Encode, Decode, Debug, Eq, PartialEq)]
+		/// Reason why a call failed.
+		#[derive(Encode, Decode, Debug, Eq, PartialEq, Clone)]
```

Comment:
Do contracts need to be able to clone? Would think not. This is probably going to increase contract size somewhat, just so that you can clone it for tests. Please move all the clone derives under the std feature as on line 24.

It might be worth considering which of the other derives should be moved as well.

Reply:
My thought process was that they would "probably" need them, also for testing but under the `std` features they will be able to. Better to have as little features enabled and based on user feedback see if something needs changing.